### PR TITLE
Add support for Pagination in Tunnels API

### PR DESCRIPTION
**Pull request by EngHabu**
Signed-off-by: Haytham Abuelfutuh <haytham@afutuh.com>

## Description

[GET Tunnels API](https://api.cloudflare.com/#cloudflare-tunnel-list-cloudflare-tunnels) supports pagination but the cloudflare-go client doesn't. I copied the way it's done from Zone Contexts into this API. Generally speaking I think all APIs that support pagination should ideally support `ReqOption`s

## Example:
```golang
actual, err := client.Tunnels(context.Background(), AccountIdentifier(testAccountID),
		TunnelListParams{}, WithPagination(PaginationOptions{PerPage: 1, Page: 2}))
```
 
## Has your change been tested?

Using the fork, we have deployed a service that syncs the status of the channels to a local DB and now is able to iterate through all tunnels we've (100+ in total)

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

**Comment by jacobbednarz** [comment]
thanks for the PR @EngHabu. this is conceptually where we want to be but the implementation is a little off with the other parts of the library. we have the [next gen client docs](https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md) that outline the way we want to handle these but in short:

- methods will have at maximum, three parameters
- pagination configuration is a part of the third and final parameter if supported
- `List` style methods automatically paginate all results unless `Page`/`PerPage` are provided
- query parameters are encoded using the struct fields, not manually

we have quite a quite methods now that implement this so check out https://github.com/cloudflare/cloudflare-go/blob/1385fe5defac70249e6c370f200215dc48940ca6/dns.go#L154 if you're after prior art.

if you can swap to these conventions, i'm 👍 on getting this one merged.

**Comment by EngHabu** [comment]
@jacobbednarz, thank you for the response. Are you ok breaking the current signature by returning a 3rd value (ResultInfo)? or should I leave this part out?

**Comment by jacobbednarz** [comment]
yep, that's totally fine here since we're moving towards the expected conventions 👍

**Comment by EngHabu** [comment]
Can you take a look now, @jacobbednarz?

File: .changelog/1206.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+tunnels: Added Pagination (through `WithPagination`) support to `Tunnels` API.
```
Comment (jacobbednarz) [inline .changelog/1206.txt]:
```suggestion
tunnels: automatically paginate `ListTunnels`
```

**Comment by jacobbednarz** [comment]
thanks @EngHabu, appreciate it!

**Pull request merged**